lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenDefinitionParticipant.java

```diff
@@ -62,27 +75,53 @@ public void findDefinition(IDefinitionRequest request, List<LocationLink> locati
 					relativeFile = new File(relativeFile, Maven.POMv4);
 				}
 				if (relativeFile.isFile()) {
-					locations.add(toLocation(relativeFile, (DOMElement)parentNode));
+					locations.add(toLocationNoRange(relativeFile, (DOMElement)parentNode));
 				}
 				return;
 			} else {
 				File relativeFile = new File(currentFolder.getParentFile(), Maven.POMv4);
 				if (match(relativeFile, dependency)) {
-					locations.add(toLocation(relativeFile, (DOMElement)parentNode));
+					locations.add(toLocationNoRange(relativeFile, (DOMElement)parentNode));
 				}
 				return;
 			}
 		}
 		if (dependency != null) {
 			File artifactLocation = getArtifactLocation(dependency);
-			LocationLink location = toLocation(artifactLocation, element);
+			LocationLink location = toLocationNoRange(artifactLocation, element);
 			if (location != null) {
 				locations.add(location);
 			}
 			return;
 		}
 	}
 
+	private LocationLink findMavenPropertyLocation(String mavenProperty, IDefinitionRequest request) {
+		DOMDocument xmlDocument = request.getXMLDocument();
+		MavenProject project = cache.getLastSuccessfulMavenProject(xmlDocument);
+		MavenProject childProj = project;
+		while (project != null && project.getProperties().containsKey(mavenProperty)) {
```

Comment:
There needs to be a check for the case that the property wasn't defined in the project, ie. the property is invalid

Reply:
As mentioned later, this is actually check before returning, so no strong need for an extra check here.

Reply:
agreed :)